src/test/allocator_tests.cpp

```diff
@@ -233,4 +235,41 @@ BOOST_AUTO_TEST_CASE(lockedpool_tests_live)
     BOOST_CHECK(pool.stats().used == initial.used);
 }
 
+BOOST_AUTO_TEST_CASE(accounting_allocation_test)
+{
+    size_t total = 0;
+    size_t dummy = 0;
+
+    {
+        typedef std::unordered_set<int, std::hash<int>, std::equal_to<int>, memusage::AccountingAllocator<int>> container_type;
+        container_type container1({}, 2, std::hash<int>(), std::equal_to<int>(), memusage::AccountingAllocator<int>(dummy)); // container1 is accounted for in 'dummy'
+        container1.insert(6);
+        {
+            container_type container2(1, std::hash<int>(), std::equal_to<int>(), memusage::AccountingAllocator<int>(total)); // container2 is accounted for in 'total'
```

Comment:
65fc229fabbc7889baca9c50ee35507543a6682e
nit - dedicated constructors could be used:
```suggestion
        container_type container1{memusage::AccountingAllocator<int>(dummy)}; // container1 is accounted for in 'dummy'
        container1.insert(6);
        {
            container_type container2{memusage::AccountingAllocator<int>(total)}; // container2 is accounted for in 'total'
```